Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1783,15 +1802,39 @@ def splunk_search_command(service):
     args = demisto.args()
 
     query = build_search_query(args)
-    search_kwargs = build_search_kwargs(args)
-    search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
-    num_of_results_from_query = search_job["resultCount"]
-
-    results_limit = float(demisto.args().get("event_limit", 100))
+    polling = argToBoolean(args.get("polling", False))
+    search_kwargs = build_search_kwargs(args, polling)
+    job_sid = args.get("sid")
+    search_job = None
+    interval_in_secs = int(args.get('interval_in_seconds', 30))
+
+    if not job_sid or not polling:
+        # create a new job to search the query.
+        search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
+        job_sid = search_job["sid"]
+        args['sid'] = job_sid
+
+    status_cmd_result = None
+    if polling:
+        status_cmd_result = splunk_job_status(service, args)
+        status = status_cmd_result.outputs['Status']
+        if status.lower() != 'done':
+            # Job is still running, schedule the next run of the command.
+            scheduled_command = schedule_polling_command("splunk-search", args, interval_in_secs)
+            status_cmd_result.scheduled_command = scheduled_command
+            status_cmd_result.readable_output = 'Job is still running, it may take a little while...'
+            return status_cmd_result
+        elif status.lower() == 'done' or search_job is None:
```

Comment:
```suggestion
        else:
```
now that this is checked only if polling is True, no need for this condition